docs/configuration/index.md

```diff
@@ -151,6 +151,35 @@ If you have 8192 mapper tasks, you could set `spark.celeborn.push.maxReqsInFligh
 
 If `celeborn.worker.flush.buffer.size` is 256 KB, we can have total slots up to 327680 slots.
 
+## Rack Awareness
+
+Celeborn can be rack-aware by setting `celeborn.client.reserveSlots.rackware.enabled` to `true` on client side.
+Shuffle partition block replica placement will use rack awareness for fault tolerance by placing one shuffle partition replica
+on a different rack. This provides data availability in the event of a network switch failure or partition within the cluster.
+
+Celeborn master daemons obtain the rack id of the cluster workers by invoking either an external script or Java class as specified by configuration files.
+Using either the Java class or external script for topology, output must adhere to the java `org.apache.hadoop.net.DNSToSwitchMapping` interface.
+The interface expects a one-to-one correspondence to be maintained and the topology information in the format of `/myrack/myhost`,
+where `/` is the topology delimiter, `myrack` is the rack identifier, and `myhost` is the individual host.
+Assuming a single `/24` subnet per rack, one could use the format of `/192.168.100.0/192.168.100.5` as a unique rack-host topology mapping.
+
+To use the Java class for topology mapping, the class name is specified by the `celeborn.hadoop.net.topology.node.switch.mapping.impl` parameter in the master configuration file.
+An example, `NetworkTopology.java`, is included with the Celeborn distribution and can be customized by the Celeborn administrator. 
+Using a Java class instead of an external script has a performance benefit in that Celeborn doesn't need to fork an external process when a new worker node registers itself.
+
+If implementing an external script, it will be specified with the `celeborn.hadoop.net.topology.script.file.name parameter` in the master side configuration files. 
```

Comment:
`celeborn.hadoop.net.topology.script.file.name parameter` is it desired or a typo?

Reply:
> `celeborn.hadoop.net.topology.script.file.name parameter` is it desired or a typo?

typo

Reply:
Fixed